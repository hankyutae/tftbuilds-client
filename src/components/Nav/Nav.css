nav,footer{
  background-color:gainsboro;
}


nav{
  position:fixed;
  top:0;
  left:0;
  right:0;
  z-index:100;
  height:30px;
  display:flex;
  align-items: center;
  justify-content:space-between;
}

.home-link{
  margin-left:10px;
}

.nav-links>a{
  margin-right:10px;
}

.nav-link{
  text-decoration:none;
}