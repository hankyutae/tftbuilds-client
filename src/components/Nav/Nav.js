import React from 'react';
import { Link } from 'react-router-dom';
import TokenService from '../../services/token-service';
import './Nav.css';

export default class Nav extends React.Component {
  state = {
    signal: this.props.signal
  }
  handleLogoutClick = () => {
    TokenService.clearAuthToken()
    this.setState({
      signal: !this.state.signal
    })
  }
  renderLogoutLink() {
    return (
      <>
        {
          this.getNavLink('Account', '/dash', ['nav-link'])
        }
        <Link className='nav-link inactive-nav-link'
          onClick={this.handleLogoutClick}
          to='/'>
          Logout
        </Link>
      </>
    )
  }
  renderLoginLink() {
    return (
      <>
        {
          this.getNavLink('Login', '/login', ['nav-link'])
        }

        {
          this.getNavLink('Register', '/signup', ['nav-link'])
        }
      </>
    )
  }
  getNavLink = (name, link, classNames) => {
    if (this.props.location.pathname === link) {
      classNames.push('active-nav-link')
    }
    else{
      classNames.push('inactive-nav-link')
    }
    return (
      <Link className={classNames.join(' ')} to={link}>
        {name}
      </Link>
    )
  }
  render() {
    return (
      <nav role="navigation">
        {
          this.getNavLink('TFTBuilds', '/', ['nav-link', 'home-link'])
        }

        <div className="nav-links">
          {
            TokenService.hasAuthToken()
              ? this.renderLogoutLink()
              : this.renderLoginLink()
          }
          {
            this.getNavLink('Your Build', '/create-build', ['nav-link','rightmost-nav-link'])
          }
        </div>
      </nav>
    );
  }
}