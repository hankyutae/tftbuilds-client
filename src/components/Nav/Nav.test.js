import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import Nav from './Nav';
import renderer from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <Nav location={{pathname:'/'}}/>
    </BrowserRouter>
  , div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create( 
      <BrowserRouter>
        <Nav location={{pathname:'/'}}/>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});