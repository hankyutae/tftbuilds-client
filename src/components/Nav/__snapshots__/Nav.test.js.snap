// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<nav
  role="navigation"
>
  <a
    className="nav-link home-link active-nav-link"
    href="/"
    onClick={[Function]}
  >
    TFTBuilds
  </a>
  <div
    className="nav-links"
  >
    <a
      className="nav-link inactive-nav-link"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
    <a
      className="nav-link inactive-nav-link"
      href="/signup"
      onClick={[Function]}
    >
      Register
    </a>
    <a
      className="nav-link rightmost-nav-link inactive-nav-link"
      href="/create-build"
      onClick={[Function]}
    >
      Your Build
    </a>
  </div>
</nav>
`;
